### PR TITLE
Wishlist page

### PR DESCRIPTION
**Pull request by tushartiwari7**
Wishlist Page 
- Add/Delete Products To/From Wishlist.
- refactored Code anad optimized it.
- Added favIcons and Dynamic Title for every Route.

### CODE REVIEW

File: src/components/ProductCard/ProductCard.jsx [hunk 1]
```diff
@@ -16,22 +17,42 @@ export const ProductCard = (product) => {
     _id,
   } = product;
   const { user, setUser } = useUser();
+
   const isInCart =
     user.isLoggedIn && user.cart.find((item) => item._id === _id);
+  const isInWishlist =
+    user.isLoggedIn && user.wishlist.find((item) => item._id === _id);
```
Comment (Bhakti199) [inline src/components/ProductCard/ProductCard.jsx]: if you are checking here for boolean value, then use `.some`

**Review by Bhakti199 — APPROVED** [review]
This code looks fine to me.
Code is written well and structured properly.
I like that idea of clustering context.
This PR is good to merge.

File: public/favicon.svg [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<svg xmlns="http://www.w3.org/2000/svg">
+  <text y="32" font-size="32">🪑</text>
```
Comment (Manju-sree) [inline public/favicon.svg:2]: Its a suggestion. you can try with different favicon, as it symbolizes only chair which is part of furniture.

File: src/components/CartItem/CartItem.jsx [hunk 3]
```diff
@@ -18,6 +23,14 @@ export const CartItem = (product) => {
     setUser((user) => ({ ...user, cart }));
   };
 
+  const addToWishListHandler = async () => {
+    const { wishlist } = await addToWishList(product);
+    setUser((user) => ({ ...user, wishlist }));
+  };
+
+  const moveToWishlistHandler = async () =>
+    await Promise.all(deleteItemHandler(), addToWishListHandler());
```
Comment (Manju-sree) [inline src/components/CartItem/CartItem.jsx:32]: good to see you have used promises.all with async fun

File: src/pages/Wishlist/Wishlist.jsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+              wishlist.map((product) => (
+                <ProductCard key={product._id} {...product} />
+              ))}
+          </div>
+        </>
+      ) : (
+        <section className="flex flex-center cart-empty">
+          <div className="flex flex-center flex-col">
+            <h3 className="h2 ubuntu text-center">
+              Login To See Items In WishList
```
Comment (Manju-sree) [inline src/pages/Wishlist/Wishlist.jsx:32]: nice.. you have used Authentication for login.

**Review by neetukumari4858 — APPROVED** [review]
looks nice folder Structure is good code is readable.it's good to merge.

**Review by Manju-sree — APPROVED** [review]
I like you have used Authentication using mockbee for wishlist and cart page. very well re-usability of code and even more readable is you code. it's fine to merge.

File: src/components/ProductCard/ProductCard.jsx (hunk 1 above)
Comment (tushartiwari7) [reply src/components/ProductCard/ProductCard.jsx]: Nice Suggestion, Implemented.

File: public/favicon.svg (hunk 2 above)
Comment (tushartiwari7) [reply public/favicon.svg:2]: okay Thanks

**Pull request merged**